applications/serial_lte_modem/src/slm_cmux.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include "slm_cmux.h"
+#include "slm_at_host.h"
+#include "slm_uart_handler.h"
+#include <zephyr/logging/log.h>
+#include <zephyr/modem/backend/uart.h>
+#include <zephyr/modem/cmux.h>
+#include <zephyr/modem/pipe.h>
+#include <assert.h>
+
+LOG_MODULE_REGISTER(slm_cmux, CONFIG_SLM_LOG_LEVEL);
+
+/* This makes use of some of the Zephyr modem subsystem that has a CMUX module implementation.
+ */
+
+static struct {
+	/* UART backend */
+	struct modem_pipe *uart_pipe;
+	struct modem_backend_uart uart_backend;
+	uint8_t uart_backend_receive_buf[512];
```

Comment:
Any reason to have these buffers as configurable? Also, these could perhaps utilize the UART buffers in future.